### DISCUSSION TITLE
Seeking co-maintainers

### DISCUSSION THREAD

**Discussion by pkgw** [tectonic-typesetting/tectonic#1122] [Announcements]
Hi folks,

You might have noticed that I've only been very limited in my engagement with the Tectonic project for most of this year. My bandwidth for open-source work outside of work hours has been decreasing steadily for years now, and if I'm honest with myself, I've been feeling a fair amount of burnout this year in particular. I still love Tectonic but nowadays my day job involves being responsible for two other substantial projects ([WorldWide Telescope](https://github.com/WorldWideTelescope) and [DASCH](https://dasch.cfa.harvard.edu) operations/pipeline analysis) and lately when the work day ends, the last thing I want to do is answer GitHub notifications.

I still have a very clear vision of where I would like Tectonic to go, but it's simply not fair to the project for me to be the single bottleneck if I can't keep up regular maintenance. So, I would like to recruit people to become co-maintainers of the project. The key tangible aspects of this status would be co-ownership of the GitHub organization, Crates.io packages, and ability to merge pull requests to the org's repositories. There are lots of other little pieces of the Tectonic project technical infrastructure and my overall goal would be to migrate to group ownership of as many of them as possible.

My ideal co-maintainers would have the following characteristics:

- Commitment to operating the project with a welcoming, constructive ethos, in line with the existing Code of Conduct. This one is really non-negotiable.
- Capacity to put substantial time into the project. For better or for worse, a lot of project maintenance is simply about putting in hours, and I want to be realistic about that.
- I'd really like to recruit at least one co-maintainer that understands LaTeX and the Tectonic codebase well and feels comfortable reviewing C/Rust pull requests. But, there are many different ways to contribute to a project, and people don't need to be a nitty-gritty tech experts to deserve leadership roles.

As you may have noticed, Tectonic doesn't currently have any sort of formalized governance model. If there are enough people interested, I would definitely be interested to move in that direction. As a reference point, I could envision seeking to formalize Tectonic as a [NumFOCUS](https://numfocus.org/) project, which would require establishing a governing committee of five people spread among at least three institutions.

I'm not exactly sure how things ought to proceed from here, but as a first step, if you might like to assume the privileges and responsibilities that come with (co-)maintainership, please send me an email at `peter@newton.cx` saying so. It'll be helpful for me to get a sense of how many people are at least interested.

Thanks all,

Peter

**Comment by bryango** [comment]
I would like to nominate @CraftSpider for commit access (if they are interested, of course), as they have already made [significant contributions](https://github.com/tectonic-typesetting/tectonic/graphs/contributors) to the codebase. The reason for me to do so is pure selfishness: I would like someone to look at my PR #1132 :laughing: and future contributions. But more importantly, I am a very happy user of tectonic myself and would like to keep it alive. Please consider!

And finally, thank you for making this wonderful program, which ease the suffering from TeX for all of us in academia. @pkgw

**Reply by CraftSpider** [reply]
Thank you for the nomination, and I appreciate the vote of confidence. I'm afraid the main reason I haven't already stepped up is because, while I'm interested in helping with Tectonic going forwards, I'm note sure I can fulfill the second bullet-point. I would love to be able to dedicate the time, but realistically, I tend to contribute to projects in an irregular capacity. I may spend many hours one week, but none for the next month. So while I intend to continue contributing, and would review PRs on an ad-hoc basis, I can't promise a reliable time commitment.

**Reply by pkgw** [reply]
@CraftSpider I'll send you an email; the only hard-requirement bullet point is the first one.

**Comment by HerrMuellerluedenscheid** [comment]
This issue brought me here: https://github.com/tectonic-typesetting/tectonic/issues/1263#issuecomment-2783411291
I worked many years with latex until ~4 years ago but a recent project brought back the need to compile latex templates from rust.
I would gladly offer a hand if another maintainer is needed.

Cheers
Marius
